instrumentation/logback/logback-mdc-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/mdc/v1_0/LoggingEventInstrumentation.java

```diff
@@ -54,11 +57,16 @@ public void transform(TypeTransformer transformer) {
 
   @SuppressWarnings("unused")
   public static class GetMdcAdvice {
+    // this has to be public otherwise the Advice cannot access it
+    // making this non-final helps greatly with testing
+    public static boolean addBaggage =
+        ConfigPropertiesUtil.getBoolean("otel.instrumentation.logback.mdc.add-baggage", false);
```

Comment:
let's extract this out into a "LogbackSingletons" class

and you can probably keep the `final` and set it during testing using `jvmArgs` in the build.gradle.kts file, e.g.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/fe263edecf52088bf4069a63fd16da0fac1dcf8d/instrumentation/logback/logback-appender-1.0/javaagent/build.gradle.kts#L54-L60

Reply:
thanks @trask where would you suggest putting the LogbackSingletons class so that it is accessible to both the javaagent and library modules?

Reply:
ignore that, I think i can configure the OpenTelemetryAppender without VM flags

Reply:
@trask I've created a pattern using testSets to cover setting the VM flag to true and false (I couldn't see an example of doing this anywhere else in the codebase).  Please let me know if there is an alternative way you'd be happier with